### ISSUE TITLE
Periodic Memory Management

### ISSUE THREAD

**Issue by pankgeorg** [fonsp/Pluto.jl#2781]
Memory management on a Garbage Collected language is something people usually don't think about. Julia is a _little_ bit different, in the sense that there isn't much memory monitoring going on the process, until the process starts feeling some memory pressure.

When the process does start to feel that memory pressure, it will aggressively run the GC _when an allocation happens_. This is not the time you'd want to run the GC, when a web request is inflight. Until julia is better at this (and _if_, because this approach is pretty good for a lot of use cases that `julia` addresses right now, e.g. doing math) we need to manage the GC a bit more manually.

**Comment by fonsp** [comment]
Nice! Do you mean on the notebook process (https://github.com/fonsp/Pluto.jl/issues/1850) or the Pluto server? cc @BioTurboNick

**Comment by fonsp** [comment]
I think a useful first step is a MWE where memory goes up and a reproducible way to measure it

**Comment by pankgeorg** [comment]
> I think a useful first step is a MWE where memory goes up and a reproducible way to measure it

Good point. The first step is adding observability, which is a whole new issue I'm going to open. Since OpenTelemetry seems to have a [memory leak](https://github.com/oolong-dev/OpenTelemetry.jl/issues/95#issue-1983567696) I wouldn't use that, but we can figure something simpler out.
